corehq/apps/enterprise/enterprise.py

```diff
@@ -216,8 +216,12 @@ def __init__(self, account, couch_user, start_date=None, end_date=None, num_days
         super().__init__(account, couch_user)
         if not end_date:
             end_date = datetime.utcnow()
+        elif isinstance(end_date, str):
+            end_date = datetime.fromisoformat(end_date)
```

Comment:
So the reason why this constructor needs to support datetime strings is because celery serializes datetime objects as strings. The current code never really passes strings to this constructor. `FormSubmissionResource`'s `get_report_task` creates `datetime` objects from the `GET` parameters. The task signature is created where both `start_date` and `end_date` are either `None` or real `datetime` objects. When celery serializes this data to be executed within the task, that's when the datetime objects are converted to their ISO string representations. Those representations should never cause an issue within the above constructor. So, in the context of this workflow, at least, `datetime.fromisoformat` will never raise an exception.

I don't think this should be an issue if we're concerned about the user supplying a faulty value to the constructor outside of this workflow, right?

Anyhow, these dates, in this workflow, originate from the string parameters from the `GET` request. If those are bad dates, we rely on tastypie's normal error handling. In production scenarios, they will receive a vague 500 error. In development scenarios, they receive the stack trace.